### PR TITLE
Certs trust support in my-resources + auto trust Burp CA

### PR DESCRIPTION
**Pull request by Skileau**
# Description
It is now possible to add a Burp CA in my-resources/burp for it to be trusted in Firefox.
For example, the CA of BurpSuitePro from the host (which will be then able to intercept requests from Firefox in Exegol).

> By default, if no CA was added to my-resouces/burp, the CA of BurpSuiteCommunity from the new container will be trusted instead.

**Note :** There are duplicates commits from PR #120 because they were required to have Burp CA trust working.

### CODE REVIEW

**Comment by ShutdownRepo** [comment]
@Dramelac the duplicate commits mean we'll need to merge this PR after #120 is merged, do you concur?

**Comment by Dramelac** [comment]
The merge will make easier to read the diff.
Depending on the situation, they might be some conflict, we will see.
I will review this PR after the merge of #120.

File: sources/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/exegol/load_supported_setups.sh:146]: Prefer using `&>` than `1>` to include stderr in the log file.

File: sources/exegol/load_supported_setups.sh
Comment (Dramelac) [inline sources/exegol/load_supported_setups.sh:153]: When the CA is generated & sign ? 
This command import the CA to Firefox right ?

File: sources/burp/conf.json
Comment (Dramelac) [inline sources/burp/conf.json:5]: Where do we specify the path of the CA to use ? There is two PATH in the setup script but I can't where / when they are generated and used by burp.

**Review by Dramelac — CHANGES_REQUESTED** [review]

File: sources/exegol/load_supported_setups.sh
Comment (Skileau) [reply sources/exegol/load_supported_setups.sh:154]: Case of Exegol's BurpSuiteCommunity :
The CA is generated when Burp is opened.
It is then downloaded to /tmp/cacert.der from where it is trusted in Firefox via certutil.

Case of host's BurpSuite :
The cacert.der in my-resources is trusted in Firefox via certutil.

File: sources/burp/conf.json
Comment (Skileau) [reply sources/burp/conf.json:5]: I added comments on every steps of the function, you can check them to have more details and then remove those you find useless.

File: sources/exegol/skel/supported_setups.md
Comment (ShutdownRepo) [inline sources/exegol/skel/supported_setups.md:46]: Can you switch from `` to ` for uniformity ?

File: sources/exegol/skel/supported_setups.md
Comment (ShutdownRepo) [inline sources/exegol/skel/supported_setups.md:41]: Leftover space at the end

**Review by ShutdownRepo — CHANGES_REQUESTED** [review]
Small changes to be done.
Also I was wondering, wouldn't make more sense to have a "certificates" folder in the firefox my-resources directory where the user should place the .DER certs to trust?
I think it would be more logical, and Burp's my-resources directory would only include changes **to** Burp.

File: sources/exegol/skel/supported_setups.md
Comment (Skileau) [reply sources/exegol/skel/supported_setups.md:46]: Oops, copy paste from .rst

**Comment by Dramelac** [comment]
> Small changes to be done.
> Also I was wondering, wouldn't make more sense to have a "certificates" folder in the firefox my-resources directory where the user should place the .DER certs to trust?
> I think it would be more logical, and Burp's my-resources directory would only include changes **to** Burp.

Agreed, if the user want to add custom CA, a dedicated directory might be helpful.
And not only DER but also PAM certificat would be great.

**Comment by Skileau** [comment]
> > Small changes to be done.
> > Also I was wondering, wouldn't make more sense to have a "certificates" folder in the firefox my-resources directory where the user should place the .DER certs to trust?
> > I think it would be more logical, and Burp's my-resources directory would only include changes **to** Burp.
> 
> Agreed, if the user want to add custom CA, a dedicated directory might be helpful.
> And not only DER but also PAM certificat would be great.

Yes we discussed it with @ShutdownRepo and the idea is now to have Firefox/certs folder which will contain all certs the user wants to trust (including Burp CA) and my-resources/burp will contain only things about burp (maybe user's extensions, and so on).

**Comment by Skileau** [comment]
I will change PR #120 by adding certs folder support and this PR (#123) will concern initialization of container's Burp Community CA if there is no Burp CA in certs folder.

**Comment by Skileau** [comment]
As discussed on Exegol's Discord, we will update this implementation to trust all of the users' CA in Firefox with my-resources and not only Burp CA.

**Comment by ShutdownRepo** [comment]
@Skileau converting PR to draft while changes are to be conducted

**Comment by ShutdownRepo** [comment]
Any news on your end @Skileau ?

**Comment by Skileau** [comment]
> Any news on this @Skileau ? 
> 
> Can you update your PR from the current dev branch (to remove the duplicated changes in the review)
> 
> 
> 
> Thank you :)

I did not have time to work on it in last months. I should be able to update it soon.

**Review by ShutdownRepo — APPROVED** [review]

**Comment by ShutdownRepo** [comment]
Docs PR ready at https://github.com/ThePorgs/Exegol-docs/pull/32

**Review by ShutdownRepo — APPROVED** [review]

**Pull request merged**
